### PR TITLE
Add postgresql to github testing workflow

### PR DESCRIPTION
**Pull request by karenc**
**Review Notes**
- Is it ok to use an environment variable `SQLALCHEMY_DATABASE_URI` to specify the database uri ~for tests~? ~(It's only for tests at the moment)~
- In the tests, the app is initialized before the database tables are created.  But during the app initialization, we try to get config from the database from a table that doesn't exist yet.  I'm catching the exception and ignoring it, but may be there's a different way of doing this
- The reason I moved the macos tests to its own job is because github returns an error when initializing the db service and I don't know how to include the db service only for ubuntu
- In /api/v1/auth/clients test, there are multiple guids returned.  When we tested with sqlite, we only checked the first guid and it always matched.  With postgresql that's not the case so I am checking all the guids.  I don't know if we're expecting multiple guids to be returned.
- I had to run `invoke app.db.upgrade` before `invoke app.db.downgrade` because the postgresql database doesn't exist and returns an error

There's more details in the commit messages, feel free to have a look.

**Pull Request Checklist**
- [x] Ensure that the PR is properly formatted
  - Example: All lint checks are passing
- [x] Ensure that the PR is properly rebased
  - Example: The PR is rebased on `develop` (commit: `<insert develop commit hash>`)
- [x] Ensure that the PR uses a consolidated database migration
  - Example: One database migration is proposed (revision `<insert develop revision>` -> `<insert new revision>`)
- [x] Ensure that the PR is properly tested
  - Example: All automated tests are passing
- [x] Ensure that the PR is properly sanitized
  - Example: No sensitive data or large content was added to this PR
- [ ] Ensure that the PR is properly reviewed

### CODE REVIEW

**Comment by bluemellophone** [comment]
What is the state of this PR?  It looks like the tests are hanging, maybe?

**Comment by karenc** [comment]
@bluemellophone I don't know what's going on with the tests.  "Testing / test (3.7) (push)" passed though

**Comment by karenc** [comment]
I'm going to rebase this PR

**Comment by karenc** [comment]
Oh oops, I'll rebase after other PRs have been merged

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -105,6 +179,8 @@ jobs:
         run: |
           source virtualenv/houston3.7/bin/activate
           pytest -s -v --gitlab-remote-login-pat "${{ secrets.GITLAB_REMOTE_LOGIN_PAT }}" --cov=./ --cov-report=xml
+        env:
+          SQLALCHEMY_DATABASE_URI: "postgresql://postgres:testing@localhost/postgres"
```
Comment (bluemellophone) [inline .github/workflows/testing.yml:183]: Are we only running the tests with PostGRESQL in this one place on Ubuntu?  I'd prefer to run tests on a full matrix of [macOS, Ubuntu] X [Python 3.7, 3.8, 3.9] X [Sqlite, PostGRES]. 

As for CodeCov reporting, we can run them on one variant, probably (Ubuntu, 3.9, PostGRES), once the tests pass above.

**Review by bluemellophone — CHANGES_REQUESTED** [review]
In-line comment about not testing PostGRES on macOS and on different versions of Python

File: .github/workflows/testing.yml (hunk 1 above)
Comment (karenc) [reply .github/workflows/testing.yml:183]: The "services" part doesn't work for macos so we can't use this way of setting up postgresql for macos, that's why I'm not including the postgresql tests for macos.

Comment (karenc) [reply .github/workflows/testing.yml:183]: I did a quick search and it looks like the only way to set up postgresql outside of github ... :/

Comment (colinwkingen) [reply .github/workflows/testing.yml]: @bluemellophone Would these tests with the addition of different python versions and Sqlite as a DB be satisfactory for this change request or do we need to investigate further for a way to include macOS?

Comment (bluemellophone) [reply .github/workflows/testing.yml]: Running PostGRES tests on Ubuntu only should be fine then, thank you @karenc for checking.  We will not be doing major deployments on macOS with PostGRES.

Comment (bluemellophone) [reply .github/workflows/testing.yml]: Then this PR is ready, I will retract this change request and likely approve.

Comment (naknomum) [reply .github/workflows/testing.yml]: i walked ben thru getting postgresql set up on his mac and it was foreign enough to me that i kinda agree with it not be necessary in terms of testing.

Comment (karenc) [reply .github/workflows/testing.yml]: The postgresql tests are failing after "invoke app.db.upgrade" so I'm looking into it.  One thing I can see is that `Submission.major_type` allows `test in the code but doesn't allow it in the migration.

**Review by bluemellophone — APPROVED** [review]
LGTM

**Review by bluemellophone — APPROVED** [review]

**Pull request merged**